activestorage/activestorage.gemspec

```diff
@@ -38,4 +38,5 @@ Gem::Specification.new do |s|
   s.add_dependency "activerecord",  version
 
   s.add_dependency "marcel",    "~> 1.0"
+  s.add_dependency "digest-crc"
```

Comment:
It would be best to avoid a new dependency that wouldn't be used by most people.

In general what we do is try to require, rescue `LoadError` etc.

Reply:
I think this can be removed as its already a dependency of the cloud service SDK gems..

Reply:
That is also not a dependency of rails. This only passes in tests since it's part of the lockfile but a fresh app with ActiveStorage will error on the eager requires.

Reply:
Is it better to do as singleton on ActiveStorage like MD5 per type? https://github.com/rails/rails/blob/9f8d180f7014b83b3eb8d3cf93fc093725b712b9/activestorage/lib/active_storage.rb#L368-L375

or is it better to memoize in ActiveStorage::Checksum class like
```
class ActiveStorage::Checksum
  require "openssl"

  attr_accessor :crc32
...
    def for(checksum_algorithm)
      case checksum_algorithm
...
      when :CRC32
        return @crc32 if @crc32
        begin
          require "digest/crc32"
        rescue LoadError
          raise LoadError, 'digest/crc32 not loaded. Please add `gem "digest-crc"` to your gemfile.'
        end
        @crc32 = Digest::CRC32
...
```

Reply:
The thing is, for services that support something better than MD5, we should just use that better thing and not rely on the global property.

e.g. google-cloud-storage can use `crc` and depends on the `digest-crc` gems, we should consider just always using that, I don't see any reason to keep using MD5 for those.

Azure it a bit more tricky because it doesn't depend on the digest crc gem.

In other words, as long as we can stop using MD5 without requiring data migration, we should do it.

Reply:
I was initially not planning to change defaults in this PR, but it makes sense if we handle cases where the stored `checksum` in the `active_storage_blobs` table doesnt match the current config for the service in this PR
Azure recommends CRC64
GCS recommends CRC32C
S3 recommends CRC64-NVMe. I submitted a [PR](https://github.com/postmodern/digest-crc/pull/45) digest/crc for add there.

Reply:
The Azure backend is deprecated and will be removed before 8.1 is released anyways, so I don't think its worth changing/thinking too hard about.